ts/navigation/MainNavigator.tsx

```diff
@@ -178,20 +180,27 @@ const navigation = createBottomTabNavigator(
       tabBarIcon: (options: { tintColor: string | null; focused: boolean }) => {
         const { routeName } = nav.state;
         const iconName: string = getIcon(routeName);
+        const messagesToRead = MessageListComponent.prototype.getMessagesToRead();
```

Comment:
can you explain how this works?

Reply:
I take from the component that manages the list of messages the number of unread messages and save them in a constant which is then used to update the view. Do you think that could be fine?

Reply:
I don't think it could possibly work, did you tested it?

Reply:
Yes, I have tested it on ios simulator and on android device and when I open a message the badge number is updated correctly.